NewmanPostman/newmantask.ts

```diff
@@ -2,50 +2,82 @@ import path = require('path');
+    let timeoutRequest = tl.getInput('timeoutRequest');
+    newman.argIf(typeof timeoutRequest != 'undefined' && timeoutRequest, ['--timeout-request', timeoutRequest]);
+    let numberOfIterations = tl.getInput('numberOfIterations');
+    newman.argIf(typeof numberOfIterations != 'undefined' && numberOfIterations, ['-n', numberOfIterations]);
+    return newman;
+}
+
 async function run() {
     try {
         tl.debug('executing newman')
         tl.setResourcePath(path.join(__dirname, 'task.json'));
 
-        var newman: trm.ToolRunner = tl.tool(tl.which('newman', true));
-
-        newman.arg('run');
-        newman.arg(tl.getPathInput('collectionFileSource', true, true));
-        newman.arg(['-e', tl.getPathInput('environment', true, true)]);
-
-        let sslClientCert = tl.getPathInput('sslClientCert', false, true);
-        newman.argIf(typeof sslClientCert != 'undefined' && sslClientCert, ['--ssl-client-cert', sslClientCert]);
-
-        let sslClientKey = tl.getPathInput('sslClientKey', false, true);
-        newman.argIf(typeof sslClientKey != 'undefined' && sslClientKey, ['--ssl-client-key', sslClientKey]);
-
-        let reporterHtmlTemplate = tl.getPathInput('reporterHtmlTemplate', false, true);
-        newman.argIf(typeof reporterHtmlTemplate != 'undefined' && reporterHtmlTemplate, ['--reporter-html-template', reporterHtmlTemplate]);
-
-        let reporterHtmlExport = tl.getInput('reporterHtmlExport');
-        newman.argIf(typeof reporterHtmlExport != 'undefined' && reporterHtmlExport, ['--reporter-html-export', reporterHtmlExport]);
+        let collectionFileSource = tl.getPathInput('collectionFileSource', true, true);
+        var taskSuccess = true;
+        if (tl.stats(collectionFileSource).isDirectory()) {
+            let contents: string[] = tl.getDelimitedInput('Contents', '\n', true);
+            collectionFileSource = path.normalize(collectionFileSource);
 
-        let reporterJsonExport = tl.getInput('reporterJsonExport');
-        newman.argIf(typeof reporterJsonExport != 'undefined' && reporterJsonExport, ['--reporter-json-export', reporterJsonExport]);
+            let allPaths: string[] = tl.find(collectionFileSource);
+            let matchedPaths: string[] = tl.match(allPaths, contents, collectionFileSource);
+            let matchedFiles: string[] = matchedPaths.filter((itemPath: string) => !tl.stats(itemPath).isDirectory());
 
-        let reporters = tl.getInput('reporters');
-        newman.argIf(typeof reporters != 'undefined' && reporters, ['-r', reporters]);
+            console.log("found %d files", matchedFiles.length);
 
-        let delayRequest = tl.getInput('delayRequest');
-        newman.argIf(typeof delayRequest != 'undefined' && delayRequest, ['--delay-request', delayRequest]);
+            if (matchedFiles.length > 0) {
+                matchedFiles.forEach((file: string) => {
+                    var newman: trm.ToolRunner = GetToolRunner();
+                    newman.arg('run');
```

Comment:
Can this line and line 50, not also be included in getting the tool runner? Seems specifying the file is the only thing different.